src/fault/Injector.cpp

```diff
@@ -84,42 +90,52 @@ bool Injector::cycleAccurateCallback(uint64_t time_ps)
                                    std::to_string(time_ps) + ")");
 #endif
     // copy pending triggers in a threadsafe manner to unknown triggers
-    if (has_pending_triggers)
+    if (unlikely(has_pending_triggers))
     {
 #if CXX0X_UP_SUPPORTED
         std::lock_guard<std::mutex> lock(sync);
 #endif
         unknown_triggers.insert(unknown_triggers.end(), pending_triggers.begin(), pending_triggers.end());
         pending_triggers.clear();
     }
+    if (unlikely(has_remove_triggers))
+    {
+#if CXX0X_UP_SUPPORTED
+        std::lock_guard<std::mutex> lock(sync);
+#endif
+        for (const auto &rm : remove_triggers)
```

Comment:
could use erase-remove idiom

    auto shouldRemove = [&](auto elem){
        return std::find(b.begin(), b.end(), elem) != b.end();
    };
    a.erase(std::remove_if(a.begin(), a.end(), shouldRemove), a.end());

otherwise the typical pattern uses a while loop and iter=a.erase(iter);

Reply:
https://github.com/tum-ei-eda/etiss/pull/100/commits/a629dff50164d34c9f76fac62dd177664842ea9b